src/main/java/com/puppycrawl/tools/checkstyle/api/FileContents.java

```diff
@@ -254,14 +257,37 @@ public TextBlock getJavadocBefore(int lineNoBefore) {
         // Lines start at 1 to the callers perspective, so need to take off 2
         int lineNo = lineNoBefore - 2;
 
-        // skip blank lines
-        while (lineNo > 0 && (lineIsBlank(lineNo) || lineIsComment(lineNo))) {
+        while (lineNo > 0 && shouldSkip(lineNo)) {
             lineNo--;
         }
 
         return javadocComments.get(lineNo);
     }
 
+    /**
+     * Checks if the specified line should be ignored.
+     * It should be ignored if it is:
+     * <ul>
+     * <li>Blank</li>
+     * <li>Line comment</li>
+     * <li>Annotation</li>
+     * </ul>
+     * @param lineNo the line number to check
+     * @return true if line should be ignored, false otherwise.
+     **/
+    private boolean shouldSkip(int lineNo) {
+        return lineIsBlank(lineNo) || lineIsComment(lineNo) || lineIsAnnotation(lineNo);
```

Comment:
please move this new methods to Check.

Attention: 
> api/FileContents.java 

this file is API, it exists only as legacy now, because till https://checkstyle.org/releasenotes.html#Release_6.0 there was no ability to have comments in tree.
I wish one day we will deprecate this Class completely or remove from it all java related parsing to force Java Checks that are AbstractCheck based to use AST, rather then manual regexp parsing.

Reply:
done